components/camel-as2/camel-as2-api/src/main/java/org/apache/camel/component/as2/api/entity/EntityParser.java

```diff
@@ -444,52 +444,56 @@ public static void parseAS2MessageEntity(HttpMessage message) throws HttpExcepti
             try {
                 // Determine Content Type of Message
                 String contentTypeStr = HttpMessageUtils.getHeaderValue(message, AS2Header.CONTENT_TYPE);
-                ContentType contentType = ContentType.parse(contentTypeStr);
-
-                // Determine Charset
-                String charsetName = StandardCharsets.US_ASCII.name();
-                Charset charset = contentType.getCharset();
-                if (charset != null) {
-                    charsetName = charset.name();
-                }
+                if (contentTypeStr != null) {
```

Comment:
Will it work too if we simply add `if (contentTypeStr == null) { return; }`?

Reply:
it works, I will add a commit with this change